server/controllers/postOrder.js

```diff
@@ -0,0 +1,35 @@
+const { insertOrder, insertOrderMeals } = require('../database/queries/order');
+const orderSchema = require('../validation/orderSchema');
+
+module.exports = (req, res, next) => {
+  let allMeals;
+  orderSchema
+    .validateAsync({
+      ...req.body, tableNumber: req.user.tableNumber,
+    })
+    .then((data) => {
+      const {
+        createdAt, totalPrice, meals,
+      } = data;
+      allMeals = meals;
+      return insertOrder(createdAt, totalPrice, req.user.tableNumber);
+    })
+    .then((order) => {
+      const { rows } = order;
+      const orderId = rows[0].id;
+      return Promise.all(allMeals.map((meal) => insertOrderMeals(orderId, meal)));
+    }).then(() => {
+      res.status(201).send({ statusCode: 201, message: 'Insertion order success' });
+    })
+    .catch((error) => {
+      if (error.code === '23503') {
```

Comment:
Depending on how many controllers you will have, you could create an error handling library so you don't have to copy and paste the logic, something like:

*a new file, errorHandlers.js*
```javascript
export const recordDoesNotExist = (error) => error && error.code === '23503';
```

*...which your controllers could then use:*

```
const { recordDoesNotExist } = require('../errorHandlers.js'); // at the top
...
    .catch((error) => {
      if (recordDoesNotExist(error)) {
```